rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1499,22 +1505,19 @@ private NodeSourcePolicy createNodeSourcePolicyActivity(NodeSourceDescriptor nod
     }
 
     private NodeSource createNodeSourceActivity(String nodeSourceName, NodeSource nodeSourceToDeploy) {
-
         try {
             nodeSourceToDeploy = PAActiveObject.turnActive(nodeSourceToDeploy, this.nodeRM);
         } catch (Exception e) {
-            logger.error(e.getMessage(), e);
-            throw new RuntimeException("Cannot create node source activity " + nodeSourceName, e);
+            String errorMessage = "Failed to create node source activity " + nodeSourceName;
+            logger.error(errorMessage, e);
+            throw new RuntimeException(errorMessage, e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)